File: 24_김수빈/2231.c
// 작성일: 2021년 10월 31일
// 작성자: 김수빈
// 프로그램명: 분해합
#include <stdio.h>

int main() {
	int num, sum, n;

	scanf_s("%d", &num);

	for (int i = 1; i < num; i++) {
		sum = n = i;

		while (n) {
			sum += n % 10;
			n /= 10;
		}

		if (num == sum) {
			printf("%d\n", i);
			return 0;
		}
	}

	printf("0\n");
	return 0;
}

File: 14_이정형/2231.py
import sys

input = sys.stdin.readline

n = int(input())

for i in range(n):
    num = sum((map(int, str(i))))  # 각 자릿수를 더하기 위해 str 변환후 map 함수 사용
    decom_num = i + num  # 분해합 = 생성자 수 + 각 자릿수의 합
    if decom_num == n:  # 이 분해합 값이 입력값과 같다면 출력 후 break
        print(i)
        break

else:
    print(0)
